snappy_wrappers/wrappers/rseqc/wrapper.py

```diff
@@ -17,63 +26,114 @@
-if [ ${{strand}} -eq -1 ]
+if [[ "$endedness" = "Single" ]]
 then
-    pattern="^This is (Pair|Single)End Data$"
-    endedness=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
-
-    pattern="^Fraction of reads failed to determine: *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
-    failed=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
-
-    if [ "$endedness" == "Single" ]
-    then
-        pattern="^Fraction of reads explained by \"\\+\\+,\\-\\-\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
-        forward=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
-        pattern="^Fraction of reads explained by \"\\+\\-,\\-\\+\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
-        reverse=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
-    else
-        pattern="^Fraction of reads explained by \"1\\+\\+,1\\-\\-,2\\+\\-,2\\-\\+\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
-        forward=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
-        pattern="^Fraction of reads explained by \"1\\+\\-,1\\-\\+,2\\+\\+,2\\-\\-\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
-        reverse=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
-    fi
+    pattern="^Fraction of reads explained by \"\\+\\+,\\-\\-\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
+    forward=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
+    pattern="^Fraction of reads explained by \"\\+\\-,\\-\\+\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
+    reverse=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
+else
+    pattern="^Fraction of reads explained by \"1\\+\\+,1\\-\\-,2\\+\\-,2\\-\\+\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
+    forward=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
+    pattern="^Fraction of reads explained by \"1\\+\\-,1\\-\\+,2\\+\\+,2\\-\\-\": *([+-]?([0-9]+(\\.[0-9]*)?|\\.[0-9]+)([EeDd][+-]?[0-9]+)?)$"
+    reverse=$(grep -E "$pattern" {snakemake.output.tsv} | sed -E "s/$pattern/\1/")
+fi
 
-    reverse=$(echo $reverse | tr '[Dde]' 'E')
+forward=$(echo $forward | tr '[Dde]' 'E')
+reverse=$(echo $reverse | tr '[Dde]' 'E')
 
-    strand=0
-    if [ $(echo "$forward > {snakemake.config[step_config][gene_expression_quantification][strandedness][threshold]}" | bc -l) -gt 0 ]
-    then
-        strand=1
-    fi
-    if [ $(echo "$reverse > {snakemake.config[step_config][gene_expression_quantification][strandedness][threshold]}" | bc -l) -gt 0 ]
-    then
-        strand=2
-    fi
+# ----- Infer protocol strandedness
+infer=0
+if [ $(echo "$forward > {config[threshold]}" | bc -l) -gt 0 ]
+then
+    infer=1
+fi
+if [ $(echo "$reverse > {config[threshold]}" | bc -l) -gt 0 ]
```

Comment:
which strand is taken if both strand counts are above the threshold? seems to be the reverse always.

Reply:
The threshold represents the proportion of reads that map in the forward or reverse direction. It must be greater than 0.5. Usually, it should be set around 0.8-0.9. If neither forward nor reverse reach this value, the protocol is assumed to be unstranded (0).